src/core/wallet/hw-wallet/ledger/apps/zil.ts

```diff
@@ -39,7 +39,7 @@ export class Zil {
                 .fromBech32Address(tx.toAddress)
                 .replace('0x', '')
                 .toLowerCase(),
-            amount: tx.amount.toString(),
+            amount: tx.amount ? tx.amount.toString() : '0',
```

Comment:
if amount is BigNumber we need to use `toFixed()` or do a new BigNumber(tx.amount).toFixed() to be sure

Reply:
on zilliqa we have fewer decimals but if someone sends a big amount it will be something like: `1.0e+15`